Files rtl8180-0.22_pre20050528.orig/ieee80211_crypt.o and rtl8180-0.22_pre20050528/ieee80211_crypt.o differ
diff -Nurp rtl8180-0.22_pre20050528.orig/ieee80211_crypt_wep.c rtl8180-0.22_pre20050528/ieee80211_crypt_wep.c
--- rtl8180-0.22_pre20050528.orig/ieee80211_crypt_wep.c	2007-02-08 23:39:39.000000000 +0000
+++ rtl8180-0.22_pre20050528/ieee80211_crypt_wep.c	2007-02-08 23:42:48.000000000 +0000
@@ -23,6 +23,7 @@
 #include <linux/slab.h>
 #include <linux/random.h>
 #include <linux/skbuff.h>
+#include <linux/mm.h>
 #include <asm/string.h>
 
 #include "ieee80211.h"
Files rtl8180-0.22_pre20050528.orig/ieee80211_module.o and rtl8180-0.22_pre20050528/ieee80211_module.o differ
Files rtl8180-0.22_pre20050528.orig/ieee80211_rx.o and rtl8180-0.22_pre20050528/ieee80211_rx.o differ
Files rtl8180-0.22_pre20050528.orig/ieee80211_tx.o and rtl8180-0.22_pre20050528/ieee80211_tx.o differ
Files rtl8180-0.22_pre20050528.orig/ieee80211_wx.o and rtl8180-0.22_pre20050528/ieee80211_wx.o differ
diff -Nurp rtl8180-0.22_pre20050528.orig/r8180_core.c rtl8180-0.22_pre20050528/r8180_core.c
--- rtl8180-0.22_pre20050528.orig/r8180_core.c	2007-02-08 23:39:39.000000000 +0000
+++ rtl8180-0.22_pre20050528/r8180_core.c	2007-02-08 23:42:25.000000000 +0000
@@ -2838,8 +2838,11 @@ short rtl8180_init(struct net_device *de
 	priv->workqueue = create_workqueue(DRV_NAME);
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,20)
 	INIT_WORK(&priv->IBSS_work,(void(*)(void*)) rtl8180_IBSS_create_wq,dev);
-	
+#else
+	INIT_WORK(&priv->IBSS_work,dev);
+#endif	
 	tasklet_init(&priv->irq_rx_tasklet,
 		     (void(*)(unsigned long)) rtl8180_irq_rx_tasklet,
 		     (unsigned long)priv);
